components/style/properties/properties.mako.rs

```diff
@@ -165,6 +165,42 @@ pub mod shorthands {
     <%include file="/shorthand/position.mako.rs" />
     <%include file="/shorthand/inherited_svg.mako.rs" />
     <%include file="/shorthand/text.mako.rs" />
+
+    // We don't defined the 'all' shorthand using the regular helpers:shorthand
+    // mechanism, since it causes some very large types to be generated.
+    <% data.declare_shorthand("all",
+                              [p.name for p in data.longhands if p.name not in ['direction', 'unicode-bidi']],
+                              spec="https://drafts.csswg.org/css-cascade-3/#all-shorthand") %>
+    pub mod all {
+        use cssparser::Parser;
+        use parser::ParserContext;
+        use properties::{ParsedDeclaration, ShorthandId, UnparsedValue};
+        use std::sync::Arc;
+
+        pub fn parse(context: &ParserContext, input: &mut Parser) -> Result<ParsedDeclaration, ()> {
+            // This function is like the parse() that is generated by
+            // helpers:shorthand, but since the only values for the 'all'
+            // shorthand when not just a single CSS-wide keyword is one
+            // with variable references, we can make this function a
+            // little simpler.
+            input.look_for_var_functions();
+            let start = input.position();
+            while let Ok(_) = input.next() {}  // Look for var()
+            if input.seen_var_functions() {
+                input.reset(start);
+                let (first_token_type, css) = try!(
+                    ::custom_properties::parse_non_custom_with_var(input));
```

Comment:
It'd be nice to share this, somehow... But I guess it's not a huge deal. Please add a comment in the relevant code pointing to this one so we can remember if we need to update it.